calendar/calendar-tool/tool/src/webapp/js/sakai-calendar.js

```diff
@@ -1,110 +1,155 @@
+    switch (defaultSubview) {
+      case 'day':
+        this.calendar.changeView('timeGridDay');
+        break;
+      case 'month':
+        this.calendar.changeView('dayGridMonth');
+        break;
+      case 'list':
+        this.calendar.changeView('listWeek');
+        break;
+      case 'week':
+      default:
+        this.calendar.changeView('timeGridWeek');
+  }
+
+    // When the user changes the view, reflect the change in a param to set the default view.
+    setDefaultCalendarSubview = () => {
+      const currentView = this.calendar.currentData.currentViewType;
+      const defaultViewParams = document.getElementsByName('calendar_default_subview');
+      if (defaultViewParams && defaultViewParams.length > 0) {
+        defaultViewParams[0].value = currentView;
+      }
+      // Reenable the button when the subview changes.
+      const changeDefaultViewButton = document.getElementsByName('eventSubmit_doDefaultview');
+      if (changeDefaultViewButton && changeDefaultViewButton.length > 0) {
+        changeDefaultViewButton[0].removeAttribute('disabled');
       }
-    }],
-    eventSourceSuccess: function(content, xhr) {
-      return content.eventArray;
-    },
-    eventBackgroundColor: eventBackgroundColor,
-    eventTextColor: eventTextColor,
-    eventContent: function (args) {
-      const isListView = calendar.getCurrentData() && calendar.getCurrentData().currentViewType.includes('list');
-      // Create the default event structure to benefit from the theme.
-      const eventTitle = document.createElement('div');
-      eventTitle.classList.add(!isListView ? 'fc-event-title' : 'fc-list-event-title');
-      // Build our event HTML including the default icon for the event type.
-      const icon = args.event._def.extendedProps.icon;
-      const title = args.event._def.title;
-      const titleElement = document.createTextNode(title);
-      const type = args.event._def.extendedProps.type;
-      const eventUrl = args.event._def.url;
-      const eventLink = document.createElement('a');
-      eventLink.href = eventUrl;
-      eventLink.title = title;
-      eventLink.appendChild(titleElement);
-      const iconSpan = document.createElement('span');
-      iconSpan.classList.add('icon');
-      iconSpan.classList.add(icon);
-      iconSpan.title = type;
-      // Append our custom event with icon to the default event structure.
-      eventTitle.appendChild(iconSpan);
-      eventTitle.appendChild(!isListView ? titleElement : eventLink);
-      // Return the modified event including out event structure.
-      return {
-        html: eventTitle.outerHTML
-      };
-    }
-  });
-  userLanguage && calendar.setOption('locale', userLanguage);
-  calendar.render();
+  }
+
+    document.querySelectorAll('.fc-timeGridWeek-button, .fc-dayGridMonth-button, .fc-timeGridDay-button, .fc-listWeek-button').forEach(viewButton => viewButton.setAttribute('onclick', 'setDefaultCalendarSubview();'));
```

Comment:
You need to wrap this baby a bit :)

Reply:
Not sure what you mean here, sorry Adrian.